.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: Apache-2.0
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master, release-1.0, release-1.1, release-1.2, release-1.3, release-1.4, release-2.0, release-2.1, release-2.2]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+  schedule:
+    - cron: '0 5 * * 6'
+
+jobs:
+  analyse:
+    name: Analyse
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+    # If this run was triggered by a pull request event, then checkout
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
+      if: ${{ github.event_name == 'pull_request' }}
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      # Override language selection by uncommenting this and choosing your languages
+      # with:
+      #   languages: go, javascript, csharp, python, cpp, java
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
+
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v1
```

Comment:
This is running `make` for no good reason and it seems to be ignoring the failure.

```
  [2020-07-27 21:04:12] [build-err] 2020/07/27 21:04:12 Makefile found, running make
```
followed by
```
  [2020-07-27 21:11:49] [build-err] fatal: ambiguous argument 'HEAD^..HEAD': unknown revision or path not in the working tree.
  [2020-07-27 21:11:49] [build-err] Use '--' to separate paths from revisions, like this:
  [2020-07-27 21:11:49] [build-err] 'git <command> [<revision>...] -- [<file>...]'
  [2020-07-27 21:11:49] [build] All files are excluded from having license headers
  [2020-07-27 21:11:49] [build] Building github.com/client9/misspell/cmd/misspell -> misspell
  [2020-07-27 21:11:49] [build-err] go: downloading github.com/client9/misspell v0.3.4
  [2020-07-27 21:11:51] [build-err] fatal: ambiguous argument 'HEAD^..HEAD': unknown revision or path not in the working tree.
  [2020-07-27 21:11:51] [build-err] Use '--' to separate paths from revisions, like this:
  [2020-07-27 21:11:51] [build-err] 'git <command> [<revision>...] -- [<file>...]'
  [2020-07-27 21:11:51] [build] Checking changed go files for spelling errors ...
```
and
```
  [2020-07-27 21:12:04] [build-err] ./scripts/golinter.sh: line 29: goimports: command not found
  [2020-07-27 21:12:04] [build-err] make: *** [linter] Error 127
  [2020-07-27 21:12:04] [build] Makefile:176: recipe for target 'linter' failed
  [2020-07-27 21:12:04] [build-err] 2020/07/27 21:12:04 Running /usr/bin/make failed, continuing anyway: exit status 2
```